### PR TITLE
Fix compilation without deprecated OpenSSL APIs

### PR DESCRIPTION
**Pull request by neheb**
First commit is for 1.0.2 and the second for 1.1.

### CODE REVIEW

File: folly/ssl/OpenSSLVersionFinder.h [hunk 1]
```diff
@@ -18,14 +18,20 @@
 #include <folly/Conv.h>
 #include <folly/portability/OpenSSL.h>
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#define OPENSSL_VERSION SSLEAY_VERSION
```
Comment (yfeldblum) [inline folly/ssl/OpenSSLVersionFinder.h]: These symbols should be namespaced to folly.

Comment (neheb) [reply folly/ssl/OpenSSLVersionFinder.h]: Not sure what you mean (I don't really know C++).

Comment (yfeldblum) [reply folly/ssl/OpenSSLVersionFinder.h]: `#define FOLLY_DETAIL_OPENSSL_VERSION ...`.

And then `#undef` them after using them (preferred) or at the bottom of the file.

Comment (Orvid) [reply folly/ssl/OpenSSLVersionFinder.h]: Prefer to have this in the OpenSSL portability header so that everything is uniform.

**Comment by neheb** [comment]
Force pushed. I think this is what was meant?

**Comment by yfeldblum** [comment]
Can this be split into two PR, one PR for each of the commits?

**Comment by yfeldblum** [comment]
Actually nah, importing as-is.

**Comment by neheb** [comment]
In case compilation fails with OpenSSL 1.0.2, folly/ssl/OpenSSLCertUtils.cpp needs to include folly/portability/OpenSSL.h . Just noticed it.

**Pull request closed without merging**